cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java

```diff
@@ -750,23 +749,38 @@ private static void writeAcdcConverterDCTerminal(String id, String conductingEqu
     }
 
-    private static void writeControlAreas(Network network, Map<Terminal, String> exportedTerminals, Map<Boundary, String> danglingLineBoundaries, String cimNamespace, String euNamespace, XMLStreamWriter writer) throws XMLStreamException {
+    private static void writeControlAreas(Network network, Map<Terminal, String> exportedTerminals, String cimNamespace, String euNamespace, XMLStreamWriter writer) throws XMLStreamException {
         CgmesControlAreas cgmesControlAreas = network.getExtension(CgmesControlAreas.class);
         for (CgmesControlArea cgmesControlArea : cgmesControlAreas.getCgmesControlAreas()) {
-            writeControlArea(cgmesControlArea, exportedTerminals, danglingLineBoundaries, cimNamespace, euNamespace, writer);
+            writeControlArea(cgmesControlArea, exportedTerminals, cimNamespace, euNamespace, writer);
         }
     }
 
-    private static void writeControlArea(CgmesControlArea cgmesControlArea, Map<Terminal, String> exportedTerminals, Map<Boundary, String> danglingLineBoundaries, String cimNamespace, String euNamespace, XMLStreamWriter writer) throws XMLStreamException {
+    private static void writeControlArea(CgmesControlArea cgmesControlArea, Map<Terminal, String> exportedTerminals, String cimNamespace, String euNamespace, XMLStreamWriter writer) throws XMLStreamException {
         ControlAreaEq.write(cgmesControlArea.getId(), cgmesControlArea.getName(), cgmesControlArea.getEnergyIdentificationCodeEIC(), cimNamespace, euNamespace, writer);
         for (Terminal terminal : cgmesControlArea.getTerminals()) {
             TieFlowEq.write(CgmesExportUtil.getUniqueId(), cgmesControlArea.getId(), exportedTerminalId(exportedTerminals, terminal), cimNamespace, writer);
         }
         for (Boundary boundary : cgmesControlArea.getBoundaries()) {
-            if (cgmesControlArea.getBoundaries().contains(boundary)) {
-                TieFlowEq.write(CgmesExportUtil.getUniqueId(), cgmesControlArea.getId(), danglingLineBoundaries.get(boundary), cimNamespace, writer);
+            TieFlowEq.write(CgmesExportUtil.getUniqueId(), cgmesControlArea.getId(), getBoundaryTerminal(boundary), cimNamespace, writer);
+        }
+    }
+
+    private static String getBoundaryTerminal(Boundary boundary) {
+        String terminalId = null;
+        Connectable<?> c = boundary.getConnectable();
+        if (c instanceof DanglingLine) {
+            terminalId = c.getAliasFromType(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "Terminal_Boundary").orElseThrow(PowsyblException::new);
+        } else {
+            int sequenceNumber = 0;
+            if (boundary.getSide().equals(Branch.Side.ONE)) {
+                sequenceNumber = 1;
+            } else if (boundary.getSide().equals(Branch.Side.TWO)) {
+                sequenceNumber = 2;
             }
+            terminalId = c.getAliasFromType(Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + CgmesNames.TERMINAL + sequenceNumber).orElseThrow(PowsyblException::new);
```

Comment:
In case of a tie line, are you sure, it should not be terminals at boundaries rather than on network side?

Reply:
Yes, the terminal is located at the boundary node in CGMES, but because both halves of the tie line lie inside the IIDM network, none of them is created as a dangling line, and the identifier of the corresponding terminal can not be obtained accessing the alias `"Terminal_Boundary"`, that only is created for dangling lines. So in this case we have to select the proper alias type.

Reply:
Yes, but we create alias of boundaries for tie lines in `ACLineSegmentConversion.convertBoundaryLines` so maybe we should retrieve that?
```java
mline.addAlias(boundaryLine1.getBoundaryTerminalId(), Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "HALF1." + CgmesNames.TERMINAL + "_Boundary");
...
mline.addAlias(boundaryLine2.getBoundaryTerminalId(), Conversion.CGMES_PREFIX_ALIAS_PROPERTIES + "HALF2." + CgmesNames.TERMINAL + "_Boundary");
```

Reply:
Oh, I see ... when the Boundary object is created the side does not refer to one of the half line sides, but is the identification of the half inside the tie line:
```java
..
} else if (identifiable instanceof TieLine) {
    side = networkContext.getReader().getAttributeValue(null, "side");
    TieLine tl = (TieLine) identifiable;
    cgmesControlArea.add(tl.getHalf(Branch.Side.valueOf(side)).getBoundary());

```
the object added as a tie flow to the control area is always the "boundary" side of one of the halves.

So you are right, we should not use the "side" at this point, but always obtain the boundary terminal from the right half line in the tie line ...

Reply:
Additionally ...

The object `Boundary` obtained for that half, that is what is stored in the `CgmesControlArea` extension, has a side. I have checked that it stores the half "number" with respect to its `TieLine` (not the location of the boundary in the half), so it is correct to build the `aliasType` using it as you propose. 

Thanks for the detailed review!

Reply:
There is an issue here:

The half lines of the tie lines present in the Network are not exported to the EQ. It is the tie line itself that is exported as a regular AC Line Segment. So, in the exported EQ file we do not have the terminals for the boundary points, only the terminals for the half lines network sides.

I will try to export the half lines separately, and their terminals, instead of the tie lines.

Reply:
Proposal: close this pull request ignoring tie flows at boundaries that are located at tie lines, and then open a new one where we export the two halves of a tie line as separate equipment. What do you think ?